src/ring_buffer.zig

```diff
@@ -276,18 +273,46 @@ const testing = std.testing;
 fn test_iterator(comptime T: type, ring: *T, values: []const u32) !void {
     const ring_index = ring.index;
 
-    var loops: usize = 0;
-    while (loops < 2) : (loops += 1) {
-        var iterator = ring.iterator();
-        var index: usize = 0;
-        while (iterator.next()) |item| {
-            try testing.expectEqual(values[index], item);
-            index += 1;
+    inline for (.{ .immutable, .mutable }) |mutability| {
+        var loops: usize = 0;
+        while (loops < 2) : (loops += 1) {
```

Comment:
While we are at it, let's also use a `for` loop here. And, given that we already do _two_ passes over the data, can we re-use this loopig for mutability testing? That is, to do something along the lines of 

```zig
                .immutable => while (iterator.next()) |item| {
                    try testing.expectEqual(values[index], if (loop == 0) values[index] else values[index] +% permutation);
                    index += 1;
                    item.* +%= permutation.
                },
```

Reply:
If we did that, it would break the other tests, since the mutable iterator tests would have different observable behavior than the immutable ones. Good catch on the for loop, will change.